File: android/supplier/DeliveryApp/app/src/main/java/hu/bme/aut/android/deliveryapp/model/Delivery.kt
package hu.bme.aut.android.deliveryapp.model

import java.io.Serializable

data class Delivery(
    val clientUser: String,
    val transporterUser: String,
    val pictureUrl: String,
    val description: String,
    val destinationLocation: Location,
    val title: String,
    val clientRating: Float,
    val pickUpUntil: String,
    val price: Float,
    val sourceLocation: Location,
    val transporterRating: Float,
    val createdAt: String,
    val _id: String,
    val pickUpFrom: String,
    var status: String,
    val updatedAt: String,
    val capacity: Capacity,
) : Serializable

File: backend/src/types/delivery.ts
import { ILocation } from './location'

export interface IDelivery {
  title: string
  description: string
  sourceLocation: ILocation
  destinationLocation: ILocation
  clientUser: string
  transporterUser?: string
  pickUpFrom: String
  pickUpUntil: String
  price: number
  clientRating?: number
  transporterRating?: number
  capacity: {
    weight: number
    height: number
    length: number
    width: number
  }
  status: string
  pictureUrl?: string
}
